ci/fireci/workflow_summary/github.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2023 Google LLC
```

Comment:
Seems like `PyGithub` does provide some of these utilities. For example 

```
/repos/{owner}/{repo}/actions/workflows/{workflow_id}/runs
GET: [github.Workflow.Workflow.get_runs()](https://pygithub.readthedocs.io/en/latest/github_objects/Workflow.html#github.Workflow.Workflow.get_runs)
```
https://pygithub.readthedocs.io/en/latest/apis.html

I could be missing some context here, why we preferred this option rather than use the api lib?

Reply:
Last time (~ 1 year ago) I checked PyGithub, it didn't have great support for Github workflows/actions. Plus, using Python3's `requests` module for REST can avoid an additional dependency.

Reply:
I just checked the PyGithub APIs, it still seems have no workflow jobs/logs related APIs.

Reply:
I Didn't realize that the were no logs related API available. Thanks for the clarification!